### ISSUE TITLE
lambda expressions are not supported in -source 1.7

### ISSUE THREAD

**Issue by tejtechtonic** [fluttercommunity/plus_plugins#523]
## System info

Issue occurs on: iOS / Android / both
Plugin name: Connectivity Plus
Plugin version: 2.0.1

## Logs

```
/Users/ankurjain/Downloads/flutter/.pub-cache/hosted/pub.dartlang.org/connectivity_plus-2.0.1/android/src/main/java/dev/fluttercommunity/plus/connectivity/ConnectivityBroadcastReceiver.java:86: error: lambda expressions are not supported in -source 1.7
    Runnable runnable = () -> events.success(connectivity.getNetworkType());
                           ^
  (use -source 8 or higher to enable lambda expressions)
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':connectivity_plus:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 33s
Exception: Gradle task assembleDebug failed with exit code 1

```

```
```

```
```

**Comment by miquelbeltran** [comment]
Seems that the issue was introduced recently in 2.0.0. In the meanwhile please use 1.0.0 I will take a look at this.

**Comment by miquelbeltran** [comment]
I can't reproduce the issue with the example project shipped in the plugin, running it for Android.

Can you provide the `build.gradle` files of your project?

The Android `build.gradle` files probably need an upgrade, check the ones in the example project, like here: https://github.com/fluttercommunity/plus_plugins/blob/main/packages/connectivity_plus/connectivity_plus/example/android/build.gradle#L8 and here: https://github.com/fluttercommunity/plus_plugins/blob/main/packages/connectivity_plus/connectivity_plus/example/android/app/build.gradle

Try to upgrade the `com.android.tools.build:gradle` and the `compileSdkVersion`

**Comment by miquelbeltran** [comment]
also paging @juliansteenbakker who worked on those changes recently

**Comment by juliansteenbakker** [comment]
This is already fixed in #521

**Comment by tejtechtonic** [comment]
> I can confirm, creating a new flutter project and adding connectivity_plus to it and the project compiles for Android.

okay thanks @miquelbeltran
